# DP: Ignore -fdebug-prefix-map in producer string (taken from the trunk)

From: bernds <bernds@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 18 Dec 2015 19:48:26 +0000
Subject: [PATCH] Ignore -fdebug-prefix-map in producer string (by Daniel Kahn
 Gillmor)

gcc/
	* dwarf2out.c (gen_producer_string): Ignore -fdebug-prefix-map.

gcc/testsuite/
	* gcc.dg/debug/dwarf2/prod-options.c: New test.



--- a/src/gcc/dwarf2out.c
+++ b/src/gcc/dwarf2out.c
@@ -21498,6 +21498,7 @@ gen_producer_string (void)
       case OPT_fpreprocessed:
       case OPT_fltrans_output_list_:
       case OPT_fresolution_:
+      case OPT_fdebug_prefix_map_:
 	/* Ignore these.  */
 	continue;
       default:
--- /dev/null
+++ b/src/gcc/testsuite/gcc.dg/debug/dwarf2/prod-options.c
@@ -0,0 +1,12 @@
+/* Verify that the DW_AT_producer does not contain certain compiler options
+   such as -fdebug-prefix-map=; this is undesirable since path names make
+   the build not reproducible.  Other skipped options could be tested here
+   as well.  */
+/* { dg-do compile } */
+/* { dg-options "-O2 -gdwarf -dA -fdebug-prefix-map=a=b" } */
+/* { dg-final { scan-assembler "DW_AT_producer: \"GNU C" } } */
+/* { dg-final { scan-assembler-not "debug-prefix-map" } } */
+
+void func (void)
+{
+}
